review/migrations/0002_test_data.py

```diff
@@ -12,25 +19,94 @@ class Migration(migrations.Migration):
+                    professional.profile_id.user_id.save()
+                    professional.profile_id.save()
+                    professional.save()
+                professionals.append(professional)
+            # Create client
+            for (username, password, first_name, last_name, email, phone_number,
+                 country, city, address) in client_test_data:
+                if User.objects.all().filter(username=username):
+                    user = User.objects.all().filter(username=username).first()
+                else:
+                    user = User.objects.create_user(username=username, password=password, first_name=first_name,
+                                                    last_name=last_name, email=email,
+                                                    last_login=datetime.datetime.now(datetime.timezone.utc))
+                if Profile.objects.all().filter(user_id=user):
+                    profile = Profile.objects.all().filter(user_id=user).first()
+                else:
+                    profile = Profile(user_id=user, user_type=UserType.Client, phone_number=phone_number,
+                                      country=country, city=city, address=address)
+                if Client.objects.all().filter(profile_id=profile):
+                    client = Client.objects.all().filter(profile_id=profile).first()
+                else:
+                    client = Client(profile_id=profile, birthday=datetime.date(2000, 1, 1))
+                    client.profile_id.user_id.save()
+                    client.profile_id.save()
+                    client.save()
+                clients.append(client)
+        # Generate review data
+        review_test_data = [
+            ('5', 'Excellent!', now - timedelta(days=10), clients[0], professionals[0]),
+            ('4', 'Good', now - timedelta(days=20), clients[1], professionals[1]),
+            ('3', 'Average', now - timedelta(days=5), clients[2], professionals[2]),
+            ('2', 'A simple test review', now - timedelta(days=2), clients[0], professionals[2]),
+            (Review.Rating.FOUR_STARS, 'Another test review', now - timedelta(days=2), clients[2], professionals[1]),
+            (Review.Rating.ONE_STAR, 'Really bad.', now - timedelta(days=2), clients[2], professionals[1]),
+        ]
+        # Create review
         with transaction.atomic():
-            for rating, description, date_posted, client_id, professional_id in test_data:
+            for rating, description, date_posted, client, professional in review_test_data:
                 Review(
                     rating=rating,
                     description=description,
                     date_posted=date_posted,
-                    client_id=client_id,
-                    professional_id=professional_id
+                    client=client,
+                    professional=professional
                 ).save()
 
     operations = [
```

Comment:
There are many things here that can be taken from the 0002_test_data file that Tal has prepared. You have written a duplicate piece of code.